standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -6995,7 +6996,17 @@ public GetValidWriteIdsResponse get_valid_write_ids(GetValidWriteIdsRequest rqst
     @Override
     public AllocateTableWriteIdsResponse allocate_table_write_ids(
             AllocateTableWriteIdsRequest rqst) throws TException {
-      return getTxnHandler().allocateTableWriteIds(rqst);
+      AllocateTableWriteIdsResponse response = getTxnHandler().allocateTableWriteIds(rqst);
+      if (listeners != null && !listeners.isEmpty()) {
+        MetaStoreListenerNotifier.notifyTxnEvent(listeners, EventType.ALLOC_WRITE_ID,
+                new AllocWriteIdEvent(rqst.getTxnIds(), rqst.getTableName(), this), null, null);
```

Comment:
AllocateWriteIdEvent should take writeIds as well to ensure idempotent behaviour in target when apply. Better to pass TxnToWriteId list. 
Also, shouldn't use the regular allocate writeid flow in target. For repl flow, need to check if the latest allocated writeid > requested writeid, then noop. If allocated writeid < requested writeid, then throw exception as it is invalid usage where they skipped some events. Also, if the table is missing in next_write_id table, then noop as table is already dropped by other event.

Reply:
ignore if its greater and fail load if its smaller